### DISCUSSION TITLE
compiling a backend only (socket) api server, api docs?

### DISCUSSION THREAD

**Discussion by dkebler** [filebrowser/filebrowser#1450] [Q&A]
I am interested in just using the backend api (with a frontend quasar app I am developing). 

So to start I'd tried to "decouple" the frontend from the backend builds as a test.  

So I got the frontend on the dev server running on a different port and proxying 8080 for filebrowser. 

with this addition to vue.config.js
```
  devServer: {
    port: 4000,
    proxy: 'http://localhost:8080'
  },
```

Per the docs I thought maybe I can compile the backend without the rice embed step and it won't embed the frontend.  So that seemed to worked and I have that running.  when loading :8080 in browser I just get a 500 error which means I think the frontend was not coupled (which is what I wanted)

But of course the proxy is not working.  Obviously I'm missing some set up in probably both the frontend and backend in order to connect the socket and to get the /api urls to proxy correctly.  

Anyway just wondering if it's possible/practical to spin up a backend only api (socket) server and if there is some documentation on running filebrowser as a backend (socket) api only. Other then rooting around in the frontend code there is no api documentation I can find.

**Comment by dkebler** [comment]
looks like the url passed to the new Websocket is probably the frontend one and not 8080.  So I might have to actaully change some code to get this to work?

```
import { removePrefix } from "./utils";
import { baseURL } from "@/utils/constants";
import store from "@/store";

const ssl = window.location.protocol === "https:";
const protocol = ssl ? "wss:" : "ws:";

export default function command(url, command, onmessage, onclose) {
  url = removePrefix(url);
  url = `${protocol}//${window.location.host}${baseURL}/api/command${url}?auth=${store.state.jwt}`;

  let conn = new window.WebSocket(url);
  conn.onopen = () => conn.send(command);
  conn.onmessage = onmessage;
  conn.onclose = onclose;
}
```

**Comment by ramiresviana** [comment]
There's no straight forward way of doing that, the project was not developed with that concept in mind. You will need to dive deep in the code base and make the needed changes. There some primordial things that you need to check:

- How the backend serves the index frontend page using templates.
- How the frontend makes requests using the current page url.
